api/product.rb

```diff
@@ -18,22 +18,59 @@
 module Api
   # Repesents a product to be managed
   class Product < Api::Object::Named
-    attr_reader :base_url
     attr_reader :objects
     attr_reader :prefix
     attr_reader :scopes
+    attr_reader :versions
 
     include Compile::Core
 
     def validate
       super
       set_variables @objects, :__product
-      check_property :base_url, String
       check_property :objects, Array
       check_property_list :objects, Api::Resource
       check_property :prefix, String
       check_property :scopes, ::Array
       check_property_list :scopes, String
+
+      check_versions
+    end
+
+    # Represents a version of the API for this product
+    class Version < Api::Object
+      attr_reader :base_url
+      attr_reader :default
+      attr_reader :name
+
+      def validate
+        super
```

Comment:
add `@default ||= false` to set @default to false if the value is not specified. This way, you avoid propagating a `nil` value.

Reply:
Done.